### ISSUE TITLE
Rule proposal: `prefer-json-import`

### ISSUE THREAD

**Issue by pereorga** [sindresorhus/eslint-plugin-unicorn#3660]
### Description

Prefer native JSON module imports for statically known JSON files. This would encourage modern ESM syntax while preserving filesystem reads where imports cannot be used.

### Examples

```js
// ❌
const data = JSON.parse(fs.readFileSync("./data.json", "utf8"));

// ✅
import data from "./data.json" with { type: "json" };
```

### Proposed rule name

prefer-json-import

### Additional Info

The rule should only report static paths and leave dynamic paths unchanged.
